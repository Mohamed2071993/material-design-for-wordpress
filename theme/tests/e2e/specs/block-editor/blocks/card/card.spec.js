/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import {
	createNewPost,
	getEditedPostContent,
	insertBlock,
} from '@wordpress/e2e-test-utils';

/**
 * Get Query without query id to match snapshot.
 *
 * @param {string} text string snapshot text.
 * @return {string} snapshot without query id.
 */
const getQueryWithoutId = text => {
	const regex = /(\"queryId\":\d{1,3}\,)/gm;
	return text.replace( regex, '' );
};

const maybe = process.env.JEST_ALLOW_SKIP_CARD ? describe.skip : describe;

maybe( 'blocks: Material card and image card', async () => {
	beforeEach( async () => {
		await createNewPost( { postType: 'page', title: `Query Page` } );
	} );

	it( 'match material card', async () => {
		await insertBlock( 'Query Loop' );

		await page.waitForSelector(
			'.block-editor-block-pattern-setup__container .wp-block-post-title'
		);

		// This inserts a pattern with mdc-card for query.
		await page.waitForSelector(
			'li.pattern-slide.active-slide[aria-label="Query with material Card"]'
		);

		// Choose the selected pattern.
		const chooseButton = await page.waitForXPath(
			'//div[contains(@class, "block-editor-block-pattern-setup__actions")]//button[text()="Choose"]'
		);

		await chooseButton.click();

		// Wait for the block to be inserted and test if mdc-card is added.
		await page.waitForSelector( 'ul li:first-child .mdc-card.post-card' );

		// Test pagination first.
		await page.waitForSelector(
			'.wp-block-material-query-pagination-first'
		);

		// Test pagination previous.
		await page.waitForSelector(
			'.wp-block-material-query-pagination-previous'
		);

		// Test pagination next.
		await page.waitForSelector(
			'.wp-block-material-query-pagination-next'
		);

		// Test pagination last.
		await page.waitForSelector(
			'.wp-block-material-query-pagination-last'
		);

		const snapshot = await getEditedPostContent();
		// Snapshot.
		expect( getQueryWithoutId( snapshot ) ).toMatchSnapshot();
	} );

	it( 'match material image card', async () => {
		await insertBlock( 'Query Loop' );

		await page.waitForSelector(
			'.block-editor-block-pattern-setup__container .wp-block-post-title'
		);

		const nextPatternButton = await page.waitForSelector(
			'.block-editor-block-pattern-setup__navigation button[aria-label="Next pattern"]'
		);
		await nextPatternButton.click();

		// This inserts a pattern with mdc image card for query.
		await page.waitForSelector(
			'li.pattern-slide.active-slide[aria-label="Query with material image card."]'
		);

		// Choose the selected pattern.
		const chooseButton = await page.waitForXPath(
			'//div[contains(@class, "block-editor-block-pattern-setup__actions")]//button[text()="Choose"]'
		);

		await chooseButton.click();

		// Wait for the block to be inserted and test if mdc-card is added.
		await page.waitForSelector(
			'ul li:first-child .wp-block-material-image-card-query'
		);

		// Test pagination first.
		await page.waitForSelector(
			'.wp-block-material-query-pagination-first'
		);

		// Test pagination previous.
		await page.waitForSelector(
			'.wp-block-material-query-pagination-previous'
		);

		// Test pagination next.
		await page.waitForSelector(
			'.wp-block-material-query-pagination-next'
		);

		// Test pagination last.
		await page.waitForSelector(
			'.wp-block-material-query-pagination-last'
		);

		// Snapshot.
		const snapshot = await getEditedPostContent();
		// Snapshot.
		expect( getQueryWithoutId( snapshot ) ).toMatchSnapshot();
	} );
} );
