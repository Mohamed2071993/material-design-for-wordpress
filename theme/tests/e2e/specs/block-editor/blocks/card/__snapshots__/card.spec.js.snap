// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks: Material card and image card match material card 1`] = `
"<!-- wp:query {\\"query\\":{\\"perPage\\":10,\\"pages\\":0,\\"offset\\":0,\\"postType\\":\\"post\\",\\"categoryIds\\":[],\\"tagIds\\":[],\\"order\\":\\"desc\\",\\"orderBy\\":\\"date\\",\\"author\\":\\"\\",\\"search\\":\\"\\",\\"exclude\\":[],\\"sticky\\":\\"\\",\\"inherit\\":true},\\"displayLayout\\":{\\"type\\":\\"flex\\",\\"columns\\":3}} -->
<div class=\\"wp-block-query\\"><!-- wp:post-template {\\"className\\":\\"is-style-material-masonry\\"} -->
<!-- wp:material/card-query /-->
<!-- /wp:post-template -->

<!-- wp:material/query-pagination {\\"layout\\":{\\"type\\":\\"flex\\",\\"justifyContent\\":\\"center\\"}} -->
<!-- wp:material/query-pagination-first /-->

<!-- wp:material/query-pagination-previous /-->

<!-- wp:material/query-pagination-next /-->

<!-- wp:material/query-pagination-last /-->
<!-- /wp:material/query-pagination --></div>
<!-- /wp:query -->"
`;

exports[`blocks: Material card and image card match material image card 1`] = `
"<!-- wp:query {\\"query\\":{\\"perPage\\":10,\\"pages\\":0,\\"offset\\":0,\\"postType\\":\\"post\\",\\"categoryIds\\":[],\\"tagIds\\":[],\\"order\\":\\"desc\\",\\"orderBy\\":\\"date\\",\\"author\\":\\"\\",\\"search\\":\\"\\",\\"exclude\\":[],\\"sticky\\":\\"\\",\\"inherit\\":true},\\"displayLayout\\":{\\"type\\":\\"flex\\",\\"columns\\":3}} -->
<div class=\\"wp-block-query\\"><!-- wp:post-template {\\"className\\":\\"is-style-material-masonry\\"} -->
<!-- wp:material/image-card-query /-->
<!-- /wp:post-template -->

<!-- wp:material/query-pagination {\\"layout\\":{\\"type\\":\\"flex\\",\\"justifyContent\\":\\"center\\"}} -->
<!-- wp:material/query-pagination-first /-->

<!-- wp:material/query-pagination-previous /-->

<!-- wp:material/query-pagination-next /-->

<!-- wp:material/query-pagination-last /-->
<!-- /wp:material/query-pagination --></div>
<!-- /wp:query -->"
`;
